ThePensionsRegulator.Umbraco.Tests/BlockListModelExtensionsTests.cs

```diff
@@ -0,0 +1,194 @@
+                UmbracoBlockListFactory.CreateBlock(
+                    UmbracoBlockListFactory.CreateContentOrSettings()
+                    .SetupUmbracoTextboxPropertyValue("MyTextProperty", "value")
+                    .Object
+                )
+            );
+
+            // Act
+            var result = BlockListModelExtensions.FindBlock(blockList, x => x.Content.GetProperty("MyTextProperty") != null);
+
+            // Assert
+            Assert.That(result, Is.EqualTo(blockList.First()));
+        }
+
+        [Test]
+        public void Block_is_matched_in_descendant_block_list()
+        {
+            var grandChildBlockList = UmbracoBlockListFactory.CreateBlockListModel(
+                UmbracoBlockListFactory.CreateBlock(
+                    UmbracoBlockListFactory.CreateContentOrSettings()
+                    .SetupUmbracoTextboxPropertyValue("MyTextProperty", "value")
+                    .Object
+                )
+            );
+
+            var childBlockList = UmbracoBlockListFactory.CreateBlockListModel(
+                UmbracoBlockListFactory.CreateBlock(
+                    UmbracoBlockListFactory.CreateContentOrSettings()
+                    .SetupUmbracoBlockListPropertyValue("grandchildBlocks", grandChildBlockList)
+                    .Object
+                    )
+                );
+
+            var parentBlockList = UmbracoBlockListFactory.CreateBlockListModel(
+                UmbracoBlockListFactory.CreateBlock(
+                    UmbracoBlockListFactory.CreateContentOrSettings()
+                    .SetupUmbracoBlockListPropertyValue("childBlocks", childBlockList)
+                    .Object
+                )
+            );
+
+            // Act
+            var result = BlockListModelExtensions.FindBlock(parentBlockList, x => x.Content.GetProperty("MyTextProperty") != null);
+
+            // Assert
+            Assert.That(result, Is.EqualTo(grandChildBlockList.First()));
+        }
+
+        [Test]
+        public void Multiple_matching_blocks_are_matched_in_descendant_block_list()
+        {
+            var blockList = CreateBlockListHierarchyWithMultipleMatchingBlocks();
+
+            // Act
+            var results = BlockListModelExtensions.FindBlocks(blockList.BlockList, x => x.Content.GetProperty("MyTextProperty") != null).ToList();
+
+            // Assert
+            Assert.That(results.Count(), Is.EqualTo(2));
+            Assert.Contains(blockList.BlocksToMatch[0], results);
+            Assert.Contains(blockList.BlocksToMatch[1], results);
+        }
+
+        private static (BlockListModel BlockList, IList<BlockListItem> BlocksToMatch) CreateBlockListHierarchyWithMultipleMatchingBlocks()
+        {
+            var matchingBlockContent1 = new Mock<IOverridablePublishedElement>();
+            matchingBlockContent1.Setup(x => x.GetProperty("MyTextProperty")).Returns(UmbracoPropertyFactory.CreateTextboxProperty("MyTextProperty", "value"));
```

Comment:
Can the literal `"MyTextProperty"` be passed made a constant that is referenced throughout these tests?

If someone changed "MyTextProperty" in one place and not others, these tests would fail giving the impression that "MyTextProperty" is important to the methods under test when what is actually important is that they are the same.